src/components/templates/InputPasswordTemplate.tsx

```diff
@@ -0,0 +1,104 @@
+import FreighterLogo from "assets/logos/freighter-logo-dark.svg";
+import { BaseLayout, BaseLayoutInsets } from "components/layout/BaseLayout";
+import Avatar from "components/sds/Avatar";
+import { Button } from "components/sds/Button";
+import { Input } from "components/sds/Input";
+import { Display, Text } from "components/sds/Typography";
+import { PASSWORD_MAX_LENGTH, PASSWORD_MIN_LENGTH } from "config/constants";
+import { px } from "helpers/dimensions";
+import useAppTranslation from "hooks/useAppTranslation";
+import React, { useCallback, useMemo, useState } from "react";
+import { View } from "react-native";
+
+interface InputPasswordTemplateProps {
+  publicKey: string | null;
+  error: string | null;
+  handleContinue: (password: string) => void;
+  isLoading: boolean;
+  handleLogout?: () => void;
+  continueButtonText?: string;
+  title?: string;
+  description?: string;
+  showLogo?: boolean;
+  insets?: BaseLayoutInsets;
+}
+
+const InputPasswordTemplate: React.FC<InputPasswordTemplateProps> = ({
+  publicKey,
+  error,
+  handleContinue,
+  isLoading,
+  handleLogout,
+  continueButtonText,
+  title,
+  description,
+  insets,
+  showLogo = true,
+}) => {
+  const { t } = useAppTranslation();
+  const [passwordValue, setPasswordValue] = useState("");
+
+  const canContinue = useMemo(
+    () =>
+      passwordValue.length >= PASSWORD_MIN_LENGTH &&
+      passwordValue.length <= PASSWORD_MAX_LENGTH,
+    [passwordValue],
+  );
+
+  const handlePasswordChange = useCallback((value: string) => {
+    setPasswordValue(value);
+  }, []);
+
+  return (
+    <BaseLayout useSafeArea useKeyboardAvoidingView insets={insets}>
+      <View className="flex-1 justify-between">
+        <View className="items-center">
+          {showLogo && <FreighterLogo width={px(48)} height={px(48)} />}
+        </View>
+        <View className="items-center justify-center bg-background-tertiary rounded-2xl p-8 gap-2">
+          <Avatar size="lg" publicAddress={publicKey ?? ""} />
+          <Display xs semiBold>
+            {title ?? t("lockScreen.title")}
+          </Display>
+          <Text
+            secondary
+            style={{
+              textAlign: "center",
```

Comment:
FYI, I think @danilosilvackl is adding a `textAlign` prop to our Text component on his PR